onnxruntime/test/autoep/library/example_plugin_ep_kernel_registry/kernels/mul.h

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "utils.h"
+#include "../../plugin_ep_utils.h"
+
+struct Mul : public OrtKernelImpl {
+ private:
+  struct PrivateTag {};
+
+ public:
+  static OrtStatus* Create(const OrtKernelInfo* info, void* state, /*out*/ std::unique_ptr<Mul>& kernel);
+
+  Mul(const OrtKernelInfo* info, void* state, PrivateTag);
+
+  static OrtStatus* ORT_API_CALL ComputeImpl(OrtKernelImpl* this_ptr, OrtKernelContext* kernel_ctx) noexcept;
+  static void ORT_API_CALL ReleaseImpl(OrtKernelImpl* this_ptr) noexcept;
```

Comment:
could potentially reduce boilerplate if we have a base kernel class.

maybe something like this.
```c++
struct BaseKernelImpl : public OrtKernelImpl {
  virtual ~BaseKernelImpl() = default;

  // implement BaseKernelImpl::ComputeImpl, BaseKernelImpl::ReleaseImpl once. they can call virtual functions.

  // derived classes implement DoCompute
  virtual OrtStatus* DoCompute(OrtKernelContext* kernel_ctx) noexcept = 0;
}
```